code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -54,18 +54,17 @@
 		kill_foam()
 		return
 
-	var/fraction = 1/initial(lifetime)
```

Comment:
Goof, you didn't even remove "carry.copy_to(chemholder, 4*carry.total_volume)" below, that is here specifically to balance the reagent reaction.